frontend/src/api/api.ts

```diff
@@ -3,17 +3,34 @@ export const api = {
   headers: {
     Accept: 'application/json',
   },
+  cache: new Map(),
+
   async get(url: string, params: Record<string, any> = {}) {
     const query = new URLSearchParams(params).toString();
+
     const response = await fetch(`${this.baseURL}${url}?${query}`, {
       headers: this.headers,
     });
 
+    if (response.status === 304) {
```

Comment:
this is not clear. why do we need this?

Reply:
While you are right that the `304` response is handled by the protocol, the issue is that the request is retried because of the `await response.json()` below in the code, so the return here is needed